exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/PcapRecordReader.java

```diff
@@ -0,0 +1,295 @@
+          minorType, majorType.getMode());
+      ValueVector vector = output.addField(field, clazz);
+      vector.allocateNew();
+      return vector;
+
+    } catch (SchemaChangeException sce) {
+      throw new NullPointerException("The addition of this field is incompatible with this OutputMutator's capabilities");
+    }
+  }
+
+  private int parsePcapFilesAndPutItToTable() throws IOException {
+    Packet packet = new Packet();
+    while (offset < validBytes) {
+
+      if (validBytes - offset < 9000) {
+        System.arraycopy(buffer, offset, buffer, 0, validBytes - offset);
+        validBytes = validBytes - offset;
+        offset = 0;
+
+        int n = in.read(buffer, validBytes, buffer.length - validBytes);
+        if (n > 0) {
+          validBytes += n;
+        }
+      }
+
+      offset = decoder.decodePacket(buffer, offset, packet);
+
+      if (addDataToTable(packet, decoder.getNetwork())) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  private boolean addDataToTable(final Packet packet, final int networkType) {
+    for (ProjectedColumnInfo pci : projectedCols) {
+      switch (pci.pcapColumn.getColumnName()) {
+        case "type":
+          setStringColumnValue(packet.getPacketType(), pci);
+          break;
+        case "timestamp":
+          setTimestampColumnValue(packet.getTimestamp(), pci);
+          break;
+        case "network":
+          setIntegerColumnValue(networkType, pci);
+          break;
+        case "src_mac_address":
+          setStringColumnValue(packet.getEthernetSource(), pci);
+          break;
+        case "dst_mac_address":
+          setStringColumnValue(packet.getEthernetDestination(), pci);
+          break;
+        case "dst_ip":
+          if (packet.getDst_ip() != null) {
+            setStringColumnValue(packet.getDst_ip().getHostAddress(), pci);
+          } else {
+            setStringColumnValue(null, pci);
+          }
+          break;
+        case "src_ip":
+          if (packet.getSrc_ip() != null) {
+            setStringColumnValue(packet.getSrc_ip().getHostAddress(), pci);
+          } else {
+            setStringColumnValue(null, pci);
+          }
+          break;
+        case "src_port":
+          setIntegerColumnValue(packet.getSrc_port(), pci);
+          break;
+        case "dst_port":
+          setIntegerColumnValue(packet.getDst_port(), pci);
+          break;
+        case "tcp_session":
+          if (packet.isTcpPacket()) {
+            setLongColumnValue(packet.getSessionHash(), pci);
+          }
+          break;
+        case "tcp_sequence":
+          if (packet.isTcpPacket()) {
+            setIntegerColumnValue(packet.getSequenceNumber(), pci);
+          }
+          break;
+        case "packet_length":
+          setIntegerColumnValue(packet.getPacketLength(), pci);
+          break;
+        case "data":
+          if (packet.getData() != null) {
+            setStringColumnValue(parseBytesToASCII(packet.getData()), pci);
```

Comment:
No. I don't want to always do this. I want the ASCII version of the packet to be an optional field that you can pull, likewise with the binary version.